oap-ape/ape-java/ape-spark/src/main/java/org/apache/spark/sql/execution/vectorized/RemoteColumnVector.java

```diff
@@ -0,0 +1,193 @@
+  }
+
+  @Override
+  public int numNulls() {
+    throw new UnsupportedOperationException("Not support yet");
+  }
+
+  @Override
+  public boolean isNullAt(int i) {
+    return !(nullBuf.getBoolean(i));
+  }
+
+
+  @Override
+  public boolean getBoolean(int i) {
+    return dataBuf.getBoolean(i);
+  }
+
+  @Override
+  public byte getByte(int i) {
+    throw new UnsupportedOperationException("Not sure about how to convert yet");
+  }
+
+  @Override
+  public short getShort(int i) {
+    throw new UnsupportedOperationException("Not sure about how to convert yet");
+  }
+
+  @Override
+  public int getInt(int i) {
+    return dataBuf.getIntLE(i * type.defaultSize());
+  }
+
+  @Override
+  public long getLong(int i) {
+    return dataBuf.getLongLE(i * type.defaultSize());
+  }
+
+  @Override
+  public float getFloat(int i) {
+    return dataBuf.getFloatLE(i * type.defaultSize());
+  }
+
+  @Override
+  public double getDouble(int i) {
+    return dataBuf.getDoubleLE(i * type.defaultSize());
+  }
+
+  @Override
+  public ColumnarArray getArray(int i) {
+    throw new UnsupportedOperationException("Not support yet");
+  }
+
+  @Override
+  public ColumnarMap getMap(int i) {
+    throw new UnsupportedOperationException("Not support yet");
+  }
+
+  @Override
+  public Decimal getDecimal(int i, int precision, int scale) {
+    if (isNullAt(i)) return null;
+    if (precision <= Decimal.MAX_INT_DIGITS()) {
+      return Decimal.createUnsafe(get32BitInt(i), precision, scale);
+    } else if (precision <= Decimal.MAX_LONG_DIGITS()) {
+      return Decimal.createUnsafe(getLong(i), precision, scale);
+    } else {
+      // convert from parquet fix length byte array to spark decimal
+      byte[] bytes = getFixedLenBinary(i);
+      BigInteger bigInteger = new BigInteger(bytes);
+      BigDecimal javaDecimal = new BigDecimal(bigInteger, scale);
+      return Decimal.apply(javaDecimal, precision, scale);
+    }
+  }
+
+  public byte[] getFixedLenBinary(int i) {
+    byte[] str = new byte[16];
+    dataBuf.getBytes(i * 16, str);
+    return str;
+  }
+
+  private int get32BitInt(int i) {
+    return dataBuf.getIntLE(i * 4);
+  }
+
+  @Override
+  public UTF8String getUTF8String(int i) {
+    byte[] str = new byte[sizes[i]];
+    dataBuf.getBytes(offsets[i], str);
+    return UTF8String.fromBytes(str);
+  }
+
+  @Override
+  public byte[] getBinary(int i) {
+    byte[] str = new byte[sizes[i]];
+    dataBuf.getBytes(offsets[i], str);
+    return str;
+  }
+
+  @Override
+  public ColumnVector getChild(int i) {
+    throw new UnsupportedOperationException("Not support yet");
+  }
+
+  public void reset() {
+    // Only element length buffer is sliced from batch response.
+    // Data buffer and null buffer will be released by response itself.
+    if (elementLengthBuf != null) {
+      elementLengthBuf.release();
+      elementLengthBuf = null;
+    }
+  }
+
+  @Override
+  public void close()  {
+    reset();
```

Comment:
Will vectors be reused over batches in Spark?
If so, will Spark call `close` before reading a new batch of data? If not, there may be a resource leak since no `release` on `elementLengthBuf`.

Reply:
add fix to release the elementLengthBuf when nextBatch invoked. For the last one, it will release in the RemoteColumnVector close() method.